### ISSUE TITLE
rr_graph: consider normalizing identifier names

### ISSUE THREAD

**Issue by mcmasterg** [verilog-to-routing/vtr-verilog-to-routing#322]
The following tags use the attribute "id":
-switch
-grid_loc
-node
-edge (as switch_id)

The following tags use the attribute "index":
-x_list/y_list
-pin

Consider using a consistent name for an identifier. Brief conversation with @kmurray he indicated would prefer id since its a more abstract representation where the user doesn't have to think as much about VPR internals

Other: there's also related ptc. It might be nice to split this into separate pin_id and channel_id, but I don't feel as strongly about it

#### Expected Behaviour

Attributes should have a consistent naming scheme

#### Current Behaviour

Attributes around identifiers are named either "id" or "index"

#### Possible Solution

Standardize them to "id"

#### Steps to Reproduce (for bugs)

Take a look at a generated rr_graph using vpr --write_rr_graph

#### Context

I'm writing rr_graph generators external to VPR (symbiflow-archs-def). It will be much easier to make this change before we start getting VPR used in production environments

#### Your Environment

* VTR revision used: 8.0.0-dev+vpr-7.0.5-5759-gc4b24af1f
* Operating System and version: N/A
* Compiler version: N/A

**Comment by kmurray** [comment]
The current `<x_list>`/`<y_list>` [index attribute](https://docs.verilogtorouting.org/en/latest/vpr/file_formats/#tag-channel-x_list) is used to identify the spatial channel location. As such I'm not sure it really should be turned into an 'id'.

We could do something like change the current format from:
```xml
<x_list index="int" info="int"/>
<y_list index="int" info="int"/>
```
to
```xml
<x_list x="int" info="int"/>
<y_list y="int" info="int"/>
```
which is more explicit about what the current `index` conceptually represents.

If we really want to fix-up the semantics (the x_list/y_list terms come from VPR's internal implementation) we could go further to something like:
```xml
<chan x="int" width="int"/>
<chan y="int" width="int"/>
```
which is a better conceptual match for what `<x_list>/<y_list>` are describing.
